Uploader.m

```diff
@@ -0,0 +1,152 @@
+#import "Uploader.h"
+
+@implementation UploadParams
+
+@end
+
+@interface Uploader()
+
+@property (copy) UploadParams* params;
+
+@property (retain) NSURLSessionDataTask* task;
+
+@end
+
+@implementation Uploader
+
+- (void)uploadFiles:(UploadParams*)params
+{
+    _params = params;
+
+    NSString *method = _params.method;
+    if ([method isEqualToString:@"POST"] || [method isEqualToString:@"PUT"]) {
+    } else {
+        method = @"POST";
+    }
+
+    NSURL *url = [NSURL URLWithString:_params.toUrl];
+    NSMutableURLRequest *req = [NSMutableURLRequest requestWithURL:url];
+    [req setHTTPMethod:method];
+
+    // set headers
+    NSString *formBoundaryString = [self generateBoundaryString];
+    NSString *contentType = [NSString stringWithFormat:@"multipart/form-data; boundary=%@", formBoundaryString];
+    [req setValue:contentType forHTTPHeaderField:@"Content-Type"];
+    for (NSString *key in _params.headers) {
+        id val = [_params.headers objectForKey:key];
+        if ([val respondsToSelector:@selector(stringValue)]) {
+            val = [val stringValue];
+        }
+        if (![val isKindOfClass:[NSString class]]) {
+            continue;
+        }
+        [req setValue:val forHTTPHeaderField:key];
+    }
+
+    NSData *formBoundaryData = [[NSString stringWithFormat:@"--%@\r\n", formBoundaryString] dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableData* reqBody = [NSMutableData data];
+
+    // add fields
+    for (NSString *key in _params.fields) {
+        id val = [_params.fields objectForKey:key];
+        if ([val respondsToSelector:@selector(stringValue)]) {
+            val = [val stringValue];
+        }
+        if (![val isKindOfClass:[NSString class]]) {
+            continue;
+        }
+
+        [reqBody appendData:formBoundaryData];
+        [reqBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"\r\n\r\n", key] dataUsingEncoding:NSUTF8StringEncoding]];
+        [reqBody appendData:[val dataUsingEncoding:NSUTF8StringEncoding]];
+        [reqBody appendData:[@"\r\n" dataUsingEncoding:NSUTF8StringEncoding]];
+    }
+
+    // add files
+    for (NSDictionary *file in _params.files) {
+        NSString *name = file[@"name"];
+        NSString *filename = file[@"filename"];
+        NSString *filepath = file[@"filepath"];
+        NSString *filetype = file[@"filetype"];
+
+        // Check if file exists
+        NSFileManager *fileManager = [NSFileManager defaultManager];
+        if (![fileManager fileExistsAtPath:filepath]){
+            // NSError* error = [NSError errorWithDomain:@"Uploader" code:NSURLErrorFileDoesNotExist userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat: @"Failed to open target file at path: %@", filepath]}];
```

Comment:
Any reason why you are swallowing this error here and just continue? I feel the user should get feedback in JS-land if an upload fails due to the file not existing.